io_scene_godot/converters/animation.py

```diff
@@ -289,22 +376,241 @@ def to_matrix(self):
+                )
+            )
+        elif (attribute == 'distance' and
+              light_node.get_type() in ('OmniLight', 'SpotLight')):
+            if light_node.get_type() == 'OmniLight':
+                track_path = base_node_path.new_copy('omni_range')
+            else:
+                track_path = base_node_path.new_copy('spot_range')
+            animation_resource.add_track(
+                build_linear_interp_value_track(
+                    track_path, fcurve, (first_frame, last_frame)
+                )
+            )
+        elif (attribute == 'spot_size' and
+              light_node.get_type() == 'SpotLight'):
+            track_path = base_node_path.new_copy('spot_angle')
+            animation_resource.add_track(
+                build_linear_interp_value_track(
+                    track_path, fcurve, (first_frame, last_frame),
+                    lambda x: math.degrees(x) / 2
+                )
+            )
+        elif (attribute == 'spot_blend' and
+              light_node.get_type() == 'SpotLight'):
+            track_path = base_node_path.new_copy('spot_angle_attenuation')
+            animation_resource.add_track(
+                build_linear_interp_value_track(
+                    track_path, fcurve, (first_frame, last_frame),
+                    lambda x: 0.2 / (x + 0.01)
+                )
+            )
+        elif attribute in ('color', 'shadow_color'):
+            if attribute not in color_frame_values_map:
+                color_frame_values_map[attribute] = [
+                    mathutils.Color()
+                    for _ in range(first_frame, last_frame)
+                ]
+            color_list = color_frame_values_map[attribute]
+            for frame in range(first_frame, last_frame):
+                color_list[frame - first_frame][
+                    fcurve.array_index] = fcurve.evaluate(frame)
+
+    for attribute, frame_value_list in color_frame_values_map.items():
+        if attribute == 'color':
+            track_path = base_node_path.new_copy('light_color')
+        else:
+            track_path = base_node_path.new_copy('shadow_color')
+
+        animation_resource.add_track(
+            Track(
+                'value',
+                track_path,
+                range(first_frame, last_frame),
+                frame_value_list
+            )
+        )
+
+
+def export_camera_action(camera_node, animation_player,
+                         blender_cam, action, animation_resource):
+    """Export camera action"""
+    if blender_cam.animation_data is None:
+        return
+
+    first_frame, last_frame = get_frame_range(action)
+    base_node_path = NodePath(
+        animation_player.parent.get_path(), camera_node.get_path()
+    )
+
+    # blender use sensor_width and f_lens to animate fov
+    # while godot directly use fov
+    fov_animated = False
+    focal_len_list = list()
+    sensor_size_list = list()
+
+    for fcurve in action.fcurves:
```

Comment:
Consider adding a new function to animation_resource something like:
```add_value_track(action, action_attribute_name, track_path, )```

Yes, you will have to iterate through action.fcurves inside that function, but it will make this whole lot very simple (and adding new ones trivial).